hera_cal/utils.py

```diff
@@ -263,7 +326,233 @@ def combine_calfits(files, fname, outdir=None, overwrite=False, broadcast_flags=
+    spw_array = np.array([0])
+    Nspws = 1
+
+    # get baselines keys
+    bls = sorted(data.bls())
+    Nbls = len(bls)
+    Nblts = Nbls * Ntimes
+
+    # reconfigure time_array and lst_array
+    time_array = np.repeat(time_array[np.newaxis], Nbls, axis=0).ravel()
+    lst_array = np.repeat(lst_array[np.newaxis], Nbls, axis=0).ravel()
+
+    # get data array
+    data_array = np.moveaxis(map(lambda p: map(lambda bl: data[str(p)][bl], bls), pols), 0, -1)
+
+    # resort time and baseline axes
+    data_array = data_array.reshape(Nblts, 1, Nfreqs, Npols)
+    nsample_array = np.ones_like(data_array, np.float)
+
+    # flags
+    if flags is None:
+        flag_array = np.zeros_like(data_array, np.float).astype(np.bool)
+    else:
+        flag_array = np.moveaxis(map(lambda p: map(lambda bl: flags[str(p)][bl].astype(np.bool), bls), pols), 0, -1)
+        flag_array = flag_array.reshape(Nblts, 1, Nfreqs, Npols)
+
+    # configure baselines
+    bls = np.repeat(np.array(bls), Ntimes, axis=0)
+
+    # get ant_1_array, ant_2_array
+    ant_1_array = bls[:,0]
+    ant_2_array = bls[:,1]
+
+    # get baseline array
+    baseline_array = 2048 * (ant_2_array+1) + (ant_1_array+1) + 2^16
+
+    # get antennas in data
+    data_ants = np.unique(np.concatenate([ant_1_array, ant_2_array]))
+    Nants_data = len(data_ants)
+
+    # get telescope ants
+    antenna_numbers = np.unique(antpos.keys())
+    Nants_telescope = len(antenna_numbers)
+    antenna_names = map(lambda a: "HH{}".format(a), antenna_numbers)
+
+    # get antpos and uvw
+    antenna_positions = np.array(map(lambda k: antpos[k], antenna_numbers))
+    uvw_array = np.array([antpos[k[0]] - antpos[k[1]] for k in zip(ant_1_array, ant_2_array)])
+
+    # get zenith location
+    zenith_dec_degrees = np.ones_like(baseline_array) * dec
+    zenith_ra_degrees = JD2RA(time_array, longitude)
+    zenith_dec = zenith_dec_degrees * np.pi / 180
+    zenith_ra = zenith_ra_degrees * np.pi / 180
+
+    # instantiate object
+    uvd = UVData()
+
+    # assign parameters
+    params = ['Nants_data', 'Nants_telescope', 'Nbls', 'Nblts', 'Nfreqs', 'Npols', 'Nspws', 'Ntimes',
+              'ant_1_array', 'ant_2_array', 'antenna_names', 'antenna_numbers', 'baseline_array',
+              'channel_width', 'data_array', 'flag_array', 'freq_array', 'history', 'instrument',
+              'integration_time', 'lst_array', 'nsample_array', 'object_name', 'phase_type',
+              'polarization_array', 'spw_array', 'telescope_location', 'telescope_name', 'time_array',
+              'uvw_array', 'vis_units', 'antenna_positions', 'zenith_dec', 'zenith_ra']              
+    for p in params:
+        uvd.__setattr__(p, locals()[p])
+
+    # write uvdata
+    if write_miriad:
+        # check output
+        fname = os.path.join(outdir, fname)
+        if os.path.exists(fname) and overwrite is False:
+            hera_cal.abscal.echo("{} exists, not overwriting".format(fname), verbose=verbose)
```

Comment:
Perhaps you don't want to do it now (if not, open an issue), but echo should probably live in utils if you want to call it in utils. Ideally, other hera_cal functions would rely on utils but not vice versa. We should probably have a good reason for making dependences circular.

Reply:
ya